### ISSUE TITLE
Syntax highlighting issue

### ISSUE THREAD

**Issue by cazgp** [joostkremers/pandoc-mode#37]
Not sure if this is pandoc-mode specific or a problem with emacs, but when loading a pandoc file, the highlighting of citations doesn't happen. In order to ensure citations are highlighted, I have to `M-x pandoc-mode` to disable, and `M-x pandoc-mode` to re-enable. This turns all @citekey blue.

Pandoc mode is being started according to the docs as: `(add-hook 'markdown-mode-hook 'pandoc-mode)`.

Ubuntu 14.04; emacs 24.5.1 (compiled); pandoc up-to-date with master; using el-get. No error messages. "Pandoc/native" in status line. The hook is at the end of `.emacs` and I tried disabling all colorschemes. Nothing else is hooking into `markdown-mode-hook`.

**Comment by joostkremers** [comment]
Yeah, I've run into this problem as well. There's something strange going on in `markdown-mode` that causes `font-lock-keywords` to be reset, but only if `markdown-mode` is set on the basis of an entry in `auto-mode-alist`. I need to take this up with the Emacs devs, but haven't gotten round to it yet...
